app/Jobs/ApplicationDeploymentJob.php

```
@@ -3659,42 +3657,124 @@ private function checkForCancellation(): void
-        // Never allow changing status from FAILED or CANCELLED_BY_USER to anything else
         if ($this->application_deployment_queue->status === ApplicationDeploymentStatus::FAILED->value) {
-            $this->application->environment->project->team?->notify(new DeploymentFailed($this->application, $this->deployment_uuid, $this->preview));
-
-            return;
+            return true;
         }
+
         if ($this->application_deployment_queue->status === ApplicationDeploymentStatus::CANCELLED_BY_USER->value) {
-            // Job was cancelled, stop execution
             $this->application_deployment_queue->addLogEntry('Deployment cancelled by user, stopping execution.');
             throw new \RuntimeException('Deployment cancelled by user', 69420);
```

PHPMD (2.15.0)

Missing class import via use statement (line '3689', column '23'). (undefined)

(MissingImport)